projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/case-flag/write-case-flag-field.component.ts

```diff
@@ -181,6 +171,32 @@ export class WriteCaseFlagFieldComponent extends AbstractFieldWriteComponent imp
     }
   }
 
+  public canMoveToFinalReviewStage(caseFlagState: CaseFlagState): boolean {
+    // If the journey is request support or manage support
+    // and the current state is either CaseFlagFieldState.FLAG_COMMENTS or CaseFlagFieldState.FLAG_UPDATE
+    // then move to final review stage
+    if (this.isDisplayContextParameterExternal) {
+      if (caseFlagState.currentCaseFlagFieldState === CaseFlagFieldState.FLAG_COMMENTS ||
+          caseFlagState.currentCaseFlagFieldState === CaseFlagFieldState.FLAG_UPDATE) {
+        return true;
+      }
+    }
+    // If the current state is CaseFlagFieldState.FLAG_STATUS then move to final review stage
```

Comment:
@johnbenjamin-hmcts - I think this could be rewritten as:
`// If the current state is one of:`
`// * CaseFlagFieldState.FLAG_STATUS`
`// * CaseFlagFieldState.FLAG_UPDATE and Welsh translation checkbox is not selected`
`// * CaseFlagFieldState.FLAG_UPDATE_WELSH_TRANSLATION`
`// then move to final review stage`
`return caseFlagState.currentCaseFlagFieldState === CaseFlagFieldState.FLAG_STATUS || (caseFlagState.currentCaseFlagFieldState === CaseFlagFieldState.FLAG_UPDATE &&
        !this.caseFlagParentFormGroup.get(CaseFlagFormFields.IS_WELSH_TRANSLATION_NEEDED)?.value) || caseFlagState.currentCaseFlagFieldState === CaseFlagFieldState.FLAG_UPDATE_WELSH_TRANSLATION;`